src/compiler/sys.ts

```diff
@@ -298,6 +298,55 @@ namespace ts {
                 };
             }
 
+            function createWatchedFileSet() {
+                const watchedDirectories = createFileMap<FileWatcher>();
+                const watchedFiles = createFileMap<(fileName: string, removed?: boolean) => void>();
+                const currentDirectory = process.cwd();
+
+                return { addFile, removeFile };
+
+                function addFile(fileName: string, callback: (fileName: string, removed?: boolean) => void): WatchedFile {
+                    const path = toPath(fileName, currentDirectory, getCanonicalPath);
+                    const parentDirPath = toPath(ts.getDirectoryPath(fileName), currentDirectory, getCanonicalPath);
+
+                    if (!watchedDirectories.contains(parentDirPath)) {
+                        watchedDirectories.set(parentDirPath, _fs.watch(
+                            parentDirPath,

```

Comment:
We currently don't detect network file systems, and I didn't find a way for it to work. Though if there is a way, we could fallback to polling in that case. In addition, as the outcome of the `fs.watch` not working reliably is failing to detect file changes, it is hard for us to know when to report to users about this state, as we don't know the ground truth.